src/ClaimToken.sol

```diff
@@ -0,0 +1,202 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.27;
+
+import {IERC20} from "@oz/token/ERC20/IERC20.sol";
+import {ECDSA} from "@oz/utils/cryptography/ECDSA.sol";
+import {Ownable} from "@oz/access/Ownable.sol";
+import {ReentrancyGuard} from "@oz/utils/ReentrancyGuard.sol";
+import {EnumerableSet} from "@oz/utils/structs/EnumerableSet.sol";
+import {MessageHashUtils} from "@oz/utils/cryptography/MessageHashUtils.sol";
+
+import {IClaimToken} from "src/interfaces/IClaimToken.sol";
+
+contract ClaimToken is IClaimToken, Ownable, ReentrancyGuard {
+    using ECDSA for bytes32;
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    // -----------------------
+    // -- Private Variables --
+    // -----------------------
+
+    mapping(address signer => bool isActivated) private _isActivatedSigner;
+    mapping(address token => mapping(bytes32 eventIDHash => mapping(address user => uint256 amount))) private
+        _userClaimedAmount;
+    mapping(address token => mapping(bytes32 eventIDHash => bool isEventOngoing)) private _isEventOngoing;
+
+    // List of signers
+    EnumerableSet.AddressSet private _signerSet;
+
+    // ---------------
+    // -- Modifiers --
+    // ---------------
+
+    // Ensures function is called by an active signer
+    modifier onlyActivatedSigner() {
+        require(_isActivatedSigner[_msgSender()], "Not an active signer");
+        _;
+    }
+
+    // -----------------
+    // -- Constructor --
+    // -----------------
+
+    constructor(address _admin, address[] memory signers) Ownable(_admin) {
```

Comment:
Check that `_admin != address(0)`

Reply:
> Check that `_admin != address(0)`

The O.Z. [Ownable contract](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/932fddf69a699a9a80fd2396fd1a2ab91cdda123/contracts/access/Ownable.sol#L39-L41) seems to filter if Admin is 0.